src/libs/search.ts

```diff
@@ -0,0 +1,190 @@
+import type {
+	ScoreAndHighlightResultItem,
+	ScoreResultItem,
+	SearchIndexes,
+	SearchResults,
+	HighlightedText
+} from '$models/search';
+
+/**
+ * Normalizes a search query by adding spaces around dots and numbers, splitting by whitespace, and removing duplicates.
+ * @param query - The search query to normalize.
+ * @returns The normalized search query.
+ */
+export function normalizeSearchQuery(query: string) {
+	return query
+		.replace(/\.([^\s])/g, '. $1')
+		.replace(/(\d+)/g, ' $1 ')
+		.split(/\s/g)
+		.filter((q) => q.length > 0)
+		.filter((q, index, self) => self.indexOf(q) === index);
+}
+
+/**
+ * Search function that takes a query string and search indexes, and returns search results.
+ * @param query - The query string to search for.
+ * @param searchIndexes - The search indexes to search in.
+ * @param k - The number of search results to return for each category. Default is 5.
+ * @param highlight - Whether to highlight the search query in the search results. Default is true.
+ * @returns The search results object containing politicians, bills, and votings categories.
+ */
+export const search = (
+	query: string,
+	searchIndexes: SearchIndexes,
+	k = 5,
```

Comment:
Even though you write a comment, I suggest using a more meaningful name eg. `maxResultPerCategory`. (A long name is fine as long as it's the most concise but still meaningful)